types/src/contract_event.rs

```diff
@@ -260,6 +321,36 @@ impl ContractEventV2 {
     pub fn event_data(&self) -> &[u8] {
         &self.event_data
     }
+
+    /// Returns the size of the event. Only used for size computation at construction time.
+    fn compute_size(type_tag: &TypeTag, event_data: &[u8]) -> anyhow::Result<usize> {
+        let size = bcs::serialized_size(&type_tag)? + event_data.len();
+        Ok(size)
+    }
+}
+
+impl<'de> Deserialize<'de> for ContractEventV2 {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where
+            D: Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        struct ContractEventPlaceholder {
+            type_tag: TypeTag,
+            #[serde(with = "serde_bytes")]
+            event_data: Vec<u8>,
+        }
+
+        let placeholder = ContractEventPlaceholder::deserialize(deserializer)?;
+        let size = Self::compute_size(&placeholder.type_tag, &placeholder.event_data)
+            .map_err(serde::de::Error::custom)?;
+
```

Comment:
and why not just call the Constructor here?

Reply:
done